File: engine/VariableAssignment.js
import {Object as InkObject} from './Object';

export class VariableAssignment extends InkObject{
	constructor(variableName, isNewDeclaration){
		super();
		this._variableName = variableName || null;
		this._isNewDeclaration = !!isNewDeclaration;
		this.isGlobal;
	}
	get variableName(){
		return this._variableName;
	}
	get isNewDeclaration(){
		return this._isNewDeclaration;
	}
	
	toString(){
		return "VarAssign to " + this.variableName;;
	}
}

File: ink_dart/lib/src/engine/variable_assignment.dart
import 'ink_object.dart';

/// An assignment to a variable in the story.
///
/// Handles both global and temporary variable assignments.
class VariableAssignment extends InkObject {
  /// The name of the variable being assigned.
  String? variableName;

  /// Whether this is a new declaration (vs re-assignment).
  bool isNewDeclaration;

  /// Whether this is a global variable.
  bool isGlobal = false;

  VariableAssignment([this.variableName, this.isNewDeclaration = false]);

  @override
  String toString() {
    final prefix = isGlobal ? 'global ' : (isNewDeclaration ? 'temp ' : '');
    return 'VarAssign(${prefix}$variableName)';
  }
}
